darts/timeseries.py

```diff
@@ -4953,43 +5032,30 @@ def _set_freq_in_xa(xa_: xr.DataArray, freq=None):
                 key.stop, (int, np.int64)
             ):
                 xa_ = self._xa.isel({self._time_dim: key})
-                if isinstance(key.step, (int, np.int64)):
-                    # new frequency is multiple of original
-                    new_freq = key.step * self.freq
-                elif key.step is None:
-                    new_freq = self.freq
-                else:
-                    new_freq = None
-                    raise_log(
-                        ValueError(
-                            f"Invalid slice step={key.step}. Only supports integer steps or `None`."
-                        ),
-                        logger=logger,
-                    )
-                # indexing may discard the freq so we restore it...
-                _set_freq_in_xa(xa_, new_freq)
+                if _get_freq(xa_) is None:
+                    # indexing may discard the freq, so we restore it...
+                    if key.step is None:
+                        freq = self.freq
+                    else:
+                        # new frequency is multiple of original
+                        freq = key.step * self.freq
+                    _set_freq_in_xa(xa_, freq)
                 return self.__class__(xa_)
             elif isinstance(key.start, pd.Timestamp) or isinstance(
                 key.stop, pd.Timestamp
             ):
                 _check_dt()
-                if isinstance(key.step, (int, np.int64)):
-                    # new frequency is multiple of original
-                    new_freq = key.step * self.freq
-                elif key.step is None:
-                    new_freq = self.freq
-                else:
-                    new_freq = None
-                    raise_log(
-                        ValueError(
-                            f"Invalid slice step={key.step}. Only supports integer steps or `None`."
-                        ),
-                        logger=logger,
-                    )
 
                 # indexing may discard the freq so we restore it...
                 xa_ = self._xa.sel({self._time_dim: key})
-                _set_freq_in_xa(xa_, new_freq)
+                if _get_freq(xa_) is None:
+                    # indexing may discard the freq, so we restore it...
+                    if key.step is None:
```

Comment:
`freq = key.step * self.freq if key.step else self.freq`

less lines and much easier to read

Reply:
👍